adapter/internal/oasparser/envoyconf/routes_configs.go

```diff
@@ -0,0 +1,360 @@
+				},
+			},
+		}
+		action.Route.RetryPolicy = commonRetryPolicy
+	}
+	action.Route.Cors = corsPolicy
+	return action
+}
+
+func generateHTTPMethodMatcher(methodRegex string, isSandbox bool, sandClusterName string) []*routev3.HeaderMatcher {
+	headerMatcher := generateHeaderMatcher(httpMethodHeader, methodRegex)
+	headerMatcherArray := []*routev3.HeaderMatcher{headerMatcher}
+	// if sandbox route, add additional header match based on cluster name header
+	if isSandbox {
+		clusterHeaderMatcher := routev3.HeaderMatcher{
+			Name: clusterHeaderName,
+			HeaderMatchSpecifier: &routev3.HeaderMatcher_StringMatch{
+				StringMatch: &envoy_type_matcherv3.StringMatcher{
+					MatchPattern: &envoy_type_matcherv3.StringMatcher_Exact{Exact: sandClusterName},
+				},
+			},
+		}
+		headerMatcherArray = append(headerMatcherArray, &clusterHeaderMatcher)
+	}
+	return headerMatcherArray
+}
+
+func generateHeaderMatcher(headerName, valueRegex string) *routev3.HeaderMatcher {
+	headerMatcherArray := &routev3.HeaderMatcher{
+		Name: headerName,
+		HeaderMatchSpecifier: &routev3.HeaderMatcher_StringMatch{
+			StringMatch: &envoy_type_matcherv3.StringMatcher{
+				MatchPattern: &envoy_type_matcherv3.StringMatcher_SafeRegex{
+					SafeRegex: &envoy_type_matcherv3.RegexMatcher{
+						EngineType: &envoy_type_matcherv3.RegexMatcher_GoogleRe2{
+							GoogleRe2: &envoy_type_matcherv3.RegexMatcher_GoogleRE2{
+								MaxProgramSize: nil,
+							},
+						},
+						Regex: "^(" + valueRegex + ")$",
+					},
+				},
+			},
+		},
+	}
+	return headerMatcherArray
+}
+
+func generateRegexMatchAndSubstitute(routePath, endpointBasePath,
+	endpointResourcePath string) *envoy_type_matcherv3.RegexMatchAndSubstitute {
+
+	substitutionString := generateSubstitutionString(endpointBasePath, endpointResourcePath)
+
+	return &envoy_type_matcherv3.RegexMatchAndSubstitute{
+		Pattern: &envoy_type_matcherv3.RegexMatcher{
+			EngineType: &envoy_type_matcherv3.RegexMatcher_GoogleRe2{
+				GoogleRe2: &envoy_type_matcherv3.RegexMatcher_GoogleRE2{
+					MaxProgramSize: nil,
+				},
+			},
+			Regex: routePath,
+		},
+		Substitution: substitutionString,
+	}
+}
+
+// Router configs for Operational Policies
+
+// generateHeaderToAddRouteConfig returns Router config for SET_HEADER
+func generateHeaderToAddRouteConfig(policyParams interface{}) (*corev3.HeaderValueOption, error) {
+	var paramsToSetHeader map[string]interface{}
+	var ok bool
+	var headerName, headerValue string
+	if paramsToSetHeader, ok = policyParams.(map[string]interface{}); !ok {
+		return nil, fmt.Errorf("Error while processing policy parameter map. Map: %v", policyParams)
+	}
+	if headerName, ok = paramsToSetHeader[constants.HeaderName].(string); !ok || headerName == "" {
+		return nil, errors.New("Policy parameter map must include headerName")
+	}
+	if headerValue, ok = paramsToSetHeader[constants.HeaderValue].(string); !ok || headerValue == "" {
+		return nil, errors.New("Policy parameter map must include headerValue")
+	}
```

Comment:
We are validating these values, before coming here. But since these are maps we have to I think, is there any possibility to convert these policies to structs?
Maybe it is fine for this PR, to use maps

Reply:
Yes, it is because of the map. I too thought of using structs, then felt that the reason you might have continued to use maps is for flexibility when a change is needed. Also we don't have to create many structs covering each policy.